mkdocs/nav.py

```diff
@@ -35,6 +69,9 @@ def __init__(self, pages_config, use_directory_urls=True):
         self.homepage = self.pages[0] if self.pages else None
         self.use_directory_urls = use_directory_urls
 
+        for page in self.pages:
+            page.url_context = self.url_context
```

Comment:
Because the page object is created at a different time, we need to add this here. It seems quite hacky tho - maybe add a method to Page to set the context?